azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/message.py

```diff
@@ -0,0 +1,53 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class Message(object):
+    def __init__(
+        self, data, creation_time_utc, message_id=None, content_encoding=None, content_type=None
```

Comment:
>creation_time_utc [](start = 20, length = 17)

why is this required?  Can't the object figure this out by calling some OS function to return the current time?

Reply:
Also: please check if this is a property we should set, or if the service does that for us?

---
In reply to: [237660405](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r237660405) [](ancestors = 237660405)

Reply:
I believe so. I included this as the Microsoft docs mention this to be explicitly set. On further thoughts I feel it means to be set from the device OS level , I shall delete this. 

---
In reply to: [237660405](https://github.com/Azure/azure-iot-sdk-python-preview/pull/27#discussion_r237660405) [](ancestors = 237660405)